persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/message/MessageDAOImpl.kt

```diff
@@ -521,8 +346,204 @@ class MessageDAOImpl(
 
     override val platformExtensions: MessageExtensions = MessageExtensionsImpl(queries, mapper, coroutineContext)
 
-    private fun MessageEntityContent.Asset.hasValidRemoteData() =
-        assetId.isNotEmpty() && assetOtrKey.isNotEmpty() && assetSha256Key.isNotEmpty()
 }
 
+internal interface MessageInsertExtension {
+    fun isValidAssetMessageUpdate(message: MessageEntity): Boolean
+    fun updateAssetMessage(message: MessageEntity)
+    fun contentTypeOf(content: MessageEntityContent): MessageEntity.ContentType
+    fun insertMessageOrIgnore(message: MessageEntity)
+}
+
+internal class MessageInsertExtensionImpl(
+    private val messagesQueries: MessagesQueries
+) : MessageInsertExtension {
```

Comment:
perhaps it would make more senes to make it an open class and extend it in the classes we are using ? I do not see a real beneift of using a delegate honestly, but I may be wrong

Reply:
The functions here are meant only to handle message insert and should not be used on it's on and functions here should not be overridden consumers have to use it as it is.

the idea is to share the logic to insert one message between MessageDAO and MigrationDAO

Reply:
I understand it purpose but the same could be achieved with internal open class right ?

Reply:
if yes in my opinion the code will look abit simpler if we just allow MessageDAO and MigrationDAO to extend it